src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -35,9 +35,10 @@ public class JavaFunctionBroker {
 	private final Map<String, ImmutablePair<String, FunctionDefinition>> methods;
 	private final ClassLoaderProvider classLoaderProvider;
 	private String workerDirectory;
-	private final AtomicBoolean oneTimeLogicInitialized = new AtomicBoolean(false);
+	private volatile boolean oneTimeLogicInitialized = false;
 	private volatile InvocationChainFactory invocationChainFactory;
 	private volatile FunctionInstanceInjector functionInstanceInjector;
+	private final Object lock = new Object();
```

Comment:
i'd give this lock a name, e.g. `initializationLock`, to make it clear it shouldn't be used for any other locking purposes